#### Importing

In [27]:
import pandas as pd
import numpy as np

#### Datasets

In [28]:
tourism_df = pd.read_csv("Tourism_Hospitality_Industry_Analysis.csv")
policy_df = pd.read_csv("Tourism_City_Policy_Data.csv")

#### Set indexes for the JOIN using Country and City as compound keys

In [22]:
tourism_df_indexed = tourism_df.set_index(['Country', 'City'])
policy_df_indexed = policy_df.set_index(['Country', 'City'])

#### Left Join

In [23]:
left_join_df = tourism_df_indexed.join(
    policy_df_indexed, 
    how="left", 
    lsuffix='_tourism', 
    rsuffix='_policy'
)

In [29]:
print(left_join_df.isna().sum())

Year                                  0
Month                                 0
Number_of_Tourists                    0
Purpose_of_Visit                      0
Average_Length_of_Stay                0
Tourist_Spending_USD                  0
Hotel_Occupancy_Rate                  0
Number_of_Hotels                      0
Hotel_Rating                          0
Average_Room_Price_USD                0
Tourism_Revenue_USD                   0
Employment_in_Tourism                 0
Contribution_to_GDP_Percent           0
Number_of_Flights                     0
Airport_Passenger_Traffic             0
Transport_Infrastructure_Quality      0
Eco_Tourism_Revenue_USD               0
Carbon_Footprint_kg                   0
Waste_Management_Rating               0
Tourist_Satisfaction_Score            0
Number_of_Online_Reviews              0
Sustainable_Tourism_Policy_Score    397
Marketing_Budget_Million_USD        397
Tourism_Tax_Rate_Percent            397
Visitor_Information_Centers         397


#### I chose a left merge becasue my main dataset is the tourism dataset and I want to keep all of its rows regardless of where there is a matching policy data. 
#### By merging on the Country and City columns, all rows from the main tourism dataset are retained, while policy data is added where available. Only the 10 matching pairs from thew policy dataset are merged.
#### NA values appear in the 5 new policy columns for the 397 unmatched tourism records. These represent cities with no policy data.

In [33]:
print(left_join_df.head())

                           Year  Month  Number_of_Tourists Purpose_of_Visit  \
Country        City                                                           
Italy          Venice      2022      2              171958         Cultural   
India          Mumbai      2020      3               55311        Religious   
Japan          Tokyo       2022      4               81551          Leisure   
United Kingdom Manchester  2021      1              160510      Educational   
Egypt          Luxor       2022     12              141387         Cultural   

                           Average_Length_of_Stay  Tourist_Spending_USD  \
Country        City                                                       
Italy          Venice                        12.4               1325.46   
India          Mumbai                        14.9               3278.67   
Japan          Tokyo                          8.1               4369.73   
United Kingdom Manchester                     4.3               2259.77

In [34]:
left_join_df.to_csv("Tourism_Hospitality_Industry_with_Policy.csv", index=True)